### DISCUSSION TITLE
[cargo-smart-release] Pre-release support

### DISCUSSION THREAD

**Discussion by jcornaz** [GitoxideLabs/gitoxide#317] [Q&A] [answered]
I see in the readme that support for pre-releases is planned. But after looking at the code, I saw some strange uses of the word "pre-release", and I am wondering what is meant by "pre-release" in the cargo-smart-release readme.

Is it "pre-release" in the [semantic-versionning](https://semver.org/#spec-item-9) sense, like `1.0.0-alpha.1`?
Or is it more [initial development phase](https://semver.org/#how-should-i-deal-with-revisions-in-the-0yz-initial-development-phase) like `0.1.0`?

I would be especially interested in support of the former (actual pre-releases, like `1.0.0-beta.1`). :-)

**Comment by Byron** [comment, marked as answer]
Thanks for bringing this up! I wouldn't be surprised if the usage of `pre-release` is ambiguous in these texts, but believe they should consistently refer to pre-1.0 crates. Maybe there is an official term for that, so the usage of the word could be fixed and made less confusing with what semver prescribes.

Settling on the official meaning here, [the code incrementing a version](https://github.com/Byron/gitoxide/blob/0dfe4a74144428e1195870a67f8aa56d1f43e1e5/cargo-smart-release/src/version.rs#L37) indeed cannot handle pre-releases yet.  There is another relevant piece of code to [determine if a version change is breaking](https://github.com/Byron/gitoxide/blob/0dfe4a74144428e1195870a67f8aa56d1f43e1e5/cargo-smart-release/src/version.rs#L180) which would probably also need adjustments.

**Reply by jcornaz** [reply]
Thanks for the answer. :)

> Maybe there is an official term for that

The semver faq names it "initial development phase". It's not really a great name. But it's less loaded than "pre-release", which is more formally defined in the semver specs.

**Reply by Byron** [reply]
Thanks for the hint! I did my best to update the wording of every usage of 'pre-release' where appropriate, and it's definitely for the better. See 498072ea42dca7b9d00bedba42829bdac92195b9 for the change.

**Reply by jcornaz** [reply]
Looks great. Thanks for the clarification.

I think you missed [this line](https://github.com/Byron/gitoxide/blob/498072ea42dca7b9d00bedba42829bdac92195b9/cargo-smart-release/README.md?plain=1#L61) which, btw also has a broken link. Probably you want the link to point to https://github.com/Byron/gitoxide/blob/main/STABILITY.md?

**Reply by Byron** [reply]
Great catch! I gave that one another shot now :D: 375cd12281297ce476df5ff9b26402356ee0ffb0 .
